apps/vtex-my-subscriptions-3/react/components/DetailsPage/BatchModal.tsx

```diff
@@ -175,24 +175,30 @@ class BatchModal extends Component<Props, State> {
     }
 
     promises?.then(() => {
-      const displayError = this.state.completed.length !== selectedIds.length
-
-      if (!displayError) {
-        showToast({ message: intl.formatMessage(messages.success) })
-        onClose()
-      } else {
-        this.setState((finalState) => {
-          finalState.completed.map((id) => delete finalState.selectionItems[id])
-
-          return {
-            selectionItems: finalState.selectionItems,
-            loading: false,
-            // If some subscription isn't on the finalState
-            // it means that some error has ocurred
-            displayError,
-          }
-        })
-      }
+      // timeout added due to state not update instantaneously
+      setTimeout(() => {
```

Comment:
I believe I used `setTimeout` as a workaround to ensure that the state is updated.
Although it works, it may be better to transform the `handleSuccess` function into a Promise, to make this state update guarantee better.
Can you test whether removing `setTimeout` and changing `handleSuccess` to a Promise works well?

```
  private handleSuccess = (id: string) => {
    return new Promise<void>((resolve) => {
      this.setState(({ completed }) => {
        completed.push(id);
        return { completed };
      }, resolve);
    });
  }
```